src/v/cloud_storage/cache_service.cc

```diff
@@ -32,13 +33,49 @@
 
 namespace cloud_storage {
 
-cache::cache(std::filesystem::path cache_dir) noexcept
+cache::cache(
+  std::filesystem::path cache_dir,
+  size_t max_cache_size,
+  ss::lowres_clock::duration check_period) noexcept
   : _cache_dir(std::move(cache_dir))
+  , _max_cache_size(max_cache_size)
+  , _check_period(check_period)
   , _cnt(0) {}
 
+ss::future<uint64_t> cache::get_curr_cache_size() {
+    // todo: implement this
+    return seastar::make_ready_future<uint64_t>(0);
+}
+
+ss::future<> cache::clean_up_cache() {
+    uint64_t curr_cache_size = co_await get_curr_cache_size();
+    if (curr_cache_size >= _max_cache_size) {
+        auto to_delete = curr_cache_size - (uint64_t)(0.8 * _max_cache_size);
+        co_await _walker.walk(_cache_dir.native());
+        auto candidates_for_deletion = _walker.files;
+        uint64_t deleted = 0;
+        while (deleted < to_delete) {
+            auto next_to_delete = candidates_for_deletion.front();
+            candidates_for_deletion.pop_front();
+            deleted += next_to_delete.size;
+            auto unused = invalidate(
+              std::filesystem::path(next_to_delete.filename));
+        }
+    }
+}
+
 ss::future<> cache::start() {
     vlog(cst_log.debug, "Starting archival cache service");
-    // todo: start maintenance cycle here
+
+    // todo: we can implement more optimal cache eviction
+    _timer.set_callback([this] {
```

Comment:
timer need to be stopped in `stop` method

Reply:
 Done.